trieste/acquisition/function.py

```diff
@@ -524,3 +527,194 @@ def prepare_acquisition_function(
+    """
+
+    def using(self, tag: str) -> BatchAcquisitionFunctionBuilder:
+        """
+        :param tag: The tag for the model, dataset pair to use to build this acquisition function.
+        :return: An acquisition function builder that selects the model and dataset specified by
+            ``tag``, as defined in :meth:`prepare_acquisition_function`.
+        """
+        single_builder = self
+
+        class _Anon(BatchAcquisitionFunctionBuilder):
+            def prepare_acquisition_function(
+                self, datasets: Mapping[str, Dataset], models: Mapping[str, ProbabilisticModel]
+            ) -> AcquisitionFunction:
+                return single_builder.prepare_acquisition_function(datasets[tag], models[tag])
+
+            def __repr__(self) -> str:
+                return f"{single_builder!r} using tag {tag!r}"
+
+        return _Anon()
+
+    @abstractmethod
+    def prepare_acquisition_function(
+        self, dataset: Dataset, model: ProbabilisticModel
+    ) -> AcquisitionFunction:
+        """
+        :param dataset: The data to use to build the acquisition function.
+        :param model: The model over the specified ``dataset``.
+        :return: An acquisition function.
+        """
+
+
+def predict_independent_f_samples_with_reparametrisation_trick(
+        model: ProbabilisticModel, at: QueryPoints, eps: tf.Tensor
+) -> tf.Tensor:
+    """
+    Returns independent samples according to the reparametrization trick.
+    The sample size S is determined by eps,
+    N is the number of query points and L is the number of latent processes.
+    :param model:
+    :param at: N query points [N, D]
+    :param eps: a [S, L] tensor of N(0, 1) i.i.d. samples
+    :return: a [S, N, L] tensor
+    """
+    assert len(at.shape) == 2
+    mean, cov = model.predict(at)   # both [N, L]
+    return mean[None, ...] + tf.math.sqrt(cov)[None, ...] * eps[:, None, :]  # [S, N, L]
+
+
+def predict_batch_f_samples_with_reparametrisation_trick(
+        model: ProbabilisticModel, at: QueryPoints, eps: tf.Tensor
+) -> tf.Tensor:
+    """
+    Returns batch-correlated samples according to the reparametrization trick.
+    The sample size S is determined by eps,
+    the batch size B must be compatible between eps and at,
+    N is the number of batches of query points
+    and L is the number of latent processes.
+    :param model:
+    :param at: N batches of query points [N, B, D]
+    :param eps: a [S, B, L] tensor of N(0, 1) i.i.d. samples
+    :return: a [S, N, B, L] tensor
+    """
+
+    assert at.shape[1] == eps.shape[1]
+
+    mean, cov = model.predict(at, full_cov=True)
+    # mean: [N, B, L]
+    # cov: [N, L, B, B]
+    mean_for_sample = tf.linalg.adjoint(mean)  # [N, L, B]
+    mean_shape = tf.shape(mean_for_sample)
+    B = mean_shape[-1]
+
+    jittermat = (
+            tf.eye(B, batch_shape=mean_shape[:-1], dtype=default_float()) * default_jitter()
```

Comment:
```suggestion
            tf.eye(B, batch_shape=mean_shape[:-1], dtype=cov.dtype) * default_jitter()
```
since we're adding it to `cov` anyway